src/main/java/uk/gov/hmcts/reform/sscs/service/TribunalsService.java

```diff
@@ -96,7 +96,12 @@ public String updateSubscription(String appealNumber, SubscriptionRequest subscr
     }
 
     public Optional<SurnameResponse> validateSurname(String appealNumber, String surname) {
-        SscsCaseData caseData = ccdService.findCcdCaseByAppealNumberAndSurname(appealNumber, surname, idamService.getIdamTokens());
+        SscsCaseData caseData = null;
+        try {
+            caseData = ccdService.findCcdCaseByAppealNumberAndSurname(appealNumber, surname, idamService.getIdamTokens());
+        } catch (uk.gov.hmcts.reform.sscs.ccd.exception.AppealNotFoundException e) {
```

Comment:
Hate this API, but we have to work around it's short-comings. It would be nice to refactor it.